lib/client.c

```diff
@@ -432,11 +437,23 @@ _snowflake_check_connection_parameters(SF_CONNECT *sf) {
         // construct a host parameter if not specified,
         char buf[1024];
         if (sf->region) {
-            sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.com",
-                     sf->account, sf->region);
+            if (strncasecmp(sf->region, "cn-", 3) == 0)
+            {
+                //region started with "cn-", use "cn" for top domain
+                sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.cn",
+                         sf->account, sf->region);
+                log_info("Connecting to CHINA Snowflake domain");
+            }
+            else
+            {
+                sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.com",
+                         sf->account, sf->region);
+                log_info("Connecting to GLOBAL Snowflake domain");
+            }
```

Comment:
Maybe we could move this check outside `if (!sf->host) {..}` condition and always check postfix of `sf->host`?
In such a case we will have only one `if .. else ...` condition and there will be no need to log similar information in `sf_ocsp.c`.

Reply:
good idea. done.